crates/cubecl-core/src/frontend/cmma.rs

```diff
@@ -210,6 +258,145 @@ impl<C: CubePrimitive> Matrix<C> {
+    /// Use [`line_layout`] to check the correct data layout for each element.
+    ///
+    /// Refer to [nvidia documentation](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#element-types-and-matrix-sizes).
+    #[allow(unused_variables)]
+    pub fn new(#[comptime] m: u32, #[comptime] n: u32, #[comptime] k: u32) -> Self {
+        intrinsic!(|scope| {
+            let ab_elem = AB::as_elem(scope);
+            let cd_elem = CD::as_elem(scope);
+
+            MmaDefinitionExpand {
+                m,
+                n,
+                k,
+                ab_elem,
+                cd_elem,
+                _ab: PhantomData,
+                _cd: PhantomData,
+            }
+        })
+    }
+
+    /// Number of elements in the matrix
+    #[allow(unused)]
+    pub fn num_elems(&self, #[comptime] ident: MatrixIdent) -> comptime_type!(u32) {
+        intrinsic!(|scope| {
+            match ident {
+                MatrixIdent::A => self.m * self.k,
+                MatrixIdent::B => self.k * self.n,
+                MatrixIdent::Accumulator => self.m * self.n,
+            }
+        })
+    }
+
+    /// Returns the number of elements handled by each lane. Should be packed into `Line`s of size
+    /// `line_size` with [`line_layout`].
+    #[allow(unused)]
+    pub fn elems_per_lane(&self, #[comptime] ident: MatrixIdent) -> comptime_type!(u32) {
+        intrinsic!(|scope| {
+            let elems = self.__expand_num_elems_method(scope, ident);
+            let plane_dim = scope.runtime_properties.mma.const_plane_size;
+            elems / plane_dim
+        })
+    }
+
+    /// The layout of each line in this matrix (row major or column major)
+    #[allow(unused)]
+    pub fn line_layout(&self, #[comptime] ident: MatrixIdent) -> comptime_type!(MatrixLayout) {
+        intrinsic!(|scope| {
+            match ident {
+                MatrixIdent::A => scope.runtime_properties.mma.register_layout_a,
+                MatrixIdent::B => scope.runtime_properties.mma.register_layout_b,
+                MatrixIdent::Accumulator => scope.runtime_properties.mma.register_layout_acc,
+            }
+        })
+    }
+
+    /// Number of elements in each line passed to the execute function
+    #[allow(unused_variables)]
+    pub fn line_size(&self, #[comptime] ident: MatrixIdent) -> comptime_type!(u32) {
+        intrinsic!(|scope| {
+            let bits = match ident {
+                MatrixIdent::A | MatrixIdent::B => Elem::size_bits(&self.ab_elem) as u32,
+                MatrixIdent::Accumulator => Elem::size_bits(&self.cd_elem) as u32,
+            };
+            let register_size = scope.runtime_properties.mma.register_size_bits;
+            // div_ceil for potential compatiblity with f64
+            register_size.div_ceil(bits)
+        })
+    }
+
+    /// Returns the coordinates of the `nth` element handled by the plane index `lane_id`
+    /// Each lane contains [`elems_per_lane`] elements in [`line_size`] chunks.
```

Comment:
It would be cool in the MmaDefinition struct to have definitions of : `lane` (Is it the plane or the unit). From that I've seems previously, the line_size is determined by the number of bits in each lane, so we could document that as well.

Reply:
I added docs to explain it to the two functions that use the term. I think we might want to eventually settle on an "official" term for the unit relative to the plane, and also use that term for the position (i.e. `LANE_POS`) instead of `UNIT_POS_PLANE` which is a bit ambiguous.